x-pack/plugins/endpoint/server/routes/endpoints.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, EndpointData } from '../types';
+import { AllEndpointsQueryBuilder } from '../services/endpoint/endpoint_query_builders';
+
+interface HitSource {
+  _source: EndpointData;
+}
+
+export interface EndpointResultList {
+  endpoints: EndpointData[];
+  total: number;
+  requestPageSize: number;
+  requestIndex: number;
+}
+
+export function registerEndpointRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  router.get(
+    {
+      path: '/api/endpoint/endpoints',
+      validate: {
+        query: schema.object({
+          pageSize: schema.number({ defaultValue: 10, min: 1 }),
```

Comment:
could you add comments explaining `pageSize` and `pageIndex` somewhere?

Reply:
I can add that, thought it was self explanatory.

Reply:
@nnamdifrankie is there an upper limit on the number of endpoints you can request in the payload?